### PR TITLE
fix: remove dependency of bindPredicate property from auth plugin into context

### PR DESCRIPTION
**Pull request by dvanmali**
# Description
Adding bind predicate as a plugin requirement causes issues when supporting [authorization passed in through context](https://neo4j.com/docs/graphql-manual/current/auth/setup/#_decoded_jwts). Default behavior according to docs performs an `all`, however, according to PR #2515, it may change.

- Removes throw error in favor of default behavior
- Creates optional bindPredicate variable available through AuthContext

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2743.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-graphql-docs-2743.surge.sh

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2743.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 43 | 58 | 1134 |
|  aggregations.NestedAggregation | 13514 | 13514 | 109 | 180 | 2174 |
|  aggregations.AggregationWithWhere | 10942 | 10942 | 82 | 144 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 20101954 | 20101954 | 3924 | 3793 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 20101954 | 20101954 | 3608 | 3678 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 8694 | 8694 | 63 | 114 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 8900 | 8900 | 65 | 107 | 2174 |
|  aggregations.DeeplyNestedCount | 10054408 | 10054408 | 5845 | 5960 | 2008534 |
|  batch-create.BatchCreate | 3600 | 3600 | 113 | 150 | 600 |
|  connect.createAndConnect | 12419 | 12419 | 427 | 363 | 3003 |
|  connections.Connection | 13042 | 13042 | 163 | 291 | 2174 |
|  connections.NestedConnection | 33883 | 33883 | 141 | 175 | 4516 |
|  connections.ConnectionWithSort | 2277 | 2277 | 78 | 96 | 1040 |
|  connections.ConnectionWithSortAndCypher | 2277 | 2277 | 236 | 381 | 1040 |
|  create.SimpleMutation | 6 | 6 | 44 | 66 | 1 |
|  cypher-directive.TopLevelCypherDirective | 99525 | 99525 | 853 | 1007 | 10168 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 127994 | 127994 | 374 | 429 | 10168 |
|  delete.SimpleDelete | 18361 | 18361 | 309 | 339 | 1040 |
|  delete.NestedDeleteInUpdate | 17123 | 17123 | 255 | 313 | 2040 |
|  query.SimpleQuery | 14120 | 14120 | 223 | 188 | 2174 |
|  query.QueryWhere | 8666 | 8675 | 43 | 66 | 2163 |
|  query.SimpleQueryWithNestedWhere | 8837 | 8846 | 94 | 69 | 2163 |
|  query.Nested | 10084290 | 10084290 | 12363 | 13218 | 2008534 |
|  query.NestedWithFilter | 37172 | 37172 | 181 | 239 | 4511 |
|  query.OrFilterOnRelationships | 34969 | 37563 | 220 | 278 | 1345 |
|  query.OrFilterOnRelationshipsAndNested | 28834 | 31309 | 401 | 386 | 1345 |
|  query.QueryWithNestedIn | 12670 | 12536 | 210 | 85 | 1324 |
|  query.NestedConnectionWhere | 8794 | 8794 | 100 | 175 | 2174 |
|  query.DeeplyNestedConnectionWhere | 8933 | 9004 | 90 | 290 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 6251 | 6251 | 155 | 321 | 1134 |
|  query.NestedWithRelationshipSingleFilters | 3021189 | 3021189 | 755 | 869 | 1003942 |
|  query.Fulltext | 96 | 96 | 32 | 54 | 16 |
|  query.FulltextWithNestedQuery | 571 | 571 | 56 | 67 | 84 |
|  sorting.SortMultipleTypes | 2493 | 2493 | 98 | 115 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1488 | 1439 | 216 | 334 | 1040 |
|  sorting.SortOnNestedFields | 13042 | 13042 | 111 | 133 | 2174 |
|  sorting.SortDeeplyNestedFields | 38965 | 38965 | 219 | 246 | 4516 |
|  sorting.SortWithTopLevelCypher | 2119 | 2119 | 159 | 234 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 76 | 89 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 135 | 82 | 35 |
|  unions.NestedUnion | 398553 | 398553 | 380 | 402 | 33033 |
|  unions.NestedUnionWithMissingFields | 372527 | 372527 | 443 | 464 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 64 | 96 | 1040 |

</details>
        
Old Schema Generation: 42.546s
Schema Generation: 41.982s

**Comment by dvanmali** [comment]
@neo4j-team-graphql @darrellwarde Changes ready for review. This issue is blocking a deployment for me. Can you please review. Thank you 😄

**Comment by dvanmali** [comment]
@Liam-Doodson

File: packages/graphql/src/translate/create-auth-and-params.ts [hunk 1]
```diff
@@ -433,10 +433,7 @@ function createRelationshipPredicate({
             authPredicate
         );
     } else {
-        if (!context.plugins?.auth) {
-            throw new Error("Auth plugin is undefined");
-        }
```
Comment (darrellwarde) [inline packages/graphql/src/translate/create-auth-and-params.ts:438]: @dvanmali, why has this check for the auth plugin been removed?

**Comment by darrellwarde** [comment]
Hey @dvanmali, sorry for the slow review on this small change.

> * Removes throw error in favor of default behavior

Why do you feel this is the right approach here? If there is no auth plugin, we have no JWT information, which would render auth functionality quite useless, which I feel should be an error case.

**Comment by dvanmali** [comment]
When we pass the [decoded JWT into the context](https://neo4j.com/docs/graphql-manual/current/auth/setup/#_decoded_jwts), auth functionality works by setting the JWT value in the context. Thus, we cannot guarantee JWT values will always be from the plugin. Hence, bindPredicate is added to the AuthContext type in this PR.

Scenarios that I can think of where JWT is passed via context is better than using the plugin:
- Cookie session authentication with an encoded JWT (prevents the need to decode the same token twice, ie better performance)
- Preprocessing a user's roles in the jwt.roles property before a Cypher query (also prevents decoding token twice)

**Review by darrellwarde — APPROVED** [review]

**Pull request merged**